test/storage/bplustree_test.cpp

```diff
@@ -0,0 +1,1468 @@
+#include <cstdlib>
+#include <set>
+#include <unordered_map>
+
+#include "storage/index/bplustree.h"
+#include "storage/storage_defs.h"
+#include "test_util/multithread_test_util.h"
+#include "test_util/test_harness.h"
+
+namespace noisepage::storage::index {
+
+unsigned int globalseed = 9;
+
+class BPlusTreeTests : public TerrierTest {
+ public:
+  const uint32_t num_threads_ = 4;
+  common::WorkerPool thread_pool_{num_threads_, {}};
+
+ protected:
+  void SetUp() override { thread_pool_.Startup(); }
+
+  void TearDown() override { thread_pool_.Shutdown(); }
+};
+
+void BasicNodeInitializationInsertReadAndFreeTest() {
+  auto bplustree = new BPlusTree<int, TupleSlot>;
+  BPlusTree<int, TupleSlot>::KeyNodePointerPair p1;
+  BPlusTree<int, TupleSlot>::KeyNodePointerPair p2;
+
+  // Get inner Node
+  auto node = BPlusTree<int, TupleSlot>::ElasticNode<BPlusTree<int, TupleSlot>::KeyNodePointerPair>::Get(
```

Comment:
Would it be better to create local variables for those magic numbers to improve readability?

Reply:
Done